services/datamanager/builtin/builtin_test.go

```diff
@@ -291,6 +291,47 @@ func TestRecoversAfterKilled(t *testing.T) {
 	test.That(t, len(mockService.getUploadedFiles()), test.ShouldEqual, 1+numArbitraryFilesToSync)
 }
 
+func TestCollectorDisabled(t *testing.T) {
+	// Register mock datasync service with a mock server.
+	rpcServer, mockService := buildAndStartLocalServer(t)
+	defer func() {
+		err := rpcServer.Stop()
+		test.That(t, err, test.ShouldBeNil)
+	}()
+
+	defer resetFolder(t, captureDir)
```

Comment:
Nit: can you do these non-deferred right above this too? If you kill a test the deferred things don't run, so it would fail on the next run in those cases

Reply:
Good catch, added